chromadb/db/duckdb.py

```diff
@@ -79,13 +79,21 @@ def get_collection_uuid_from_name(self, name):
     #
     #  COLLECTION METHODS
     #
-    def create_collection(self, name, metadata=None):
+    def create_collection(self, name, metadata=None, get_or_create=False):
         if metadata is None:
             metadata = {}
 
         # poor man's unique constraint
-        if not self.get_collection(name).empty:
-            raise Exception(f"collection with name {name} already exists")
+        dupe_check = self.get_collection(name)
```

Comment:
Storing this as a variable saves us one extra query on line 93.

Reply:
We also need to update this, as it doesn't check for the metadata. 

For example, collection `test` might exist with `embedding_function=fn1`. If I try to create a collection `test` with `embedding_function=fn2`, it should NOT return the existing one, but it also can't create one with the same name. We should figure out what we want to do here.

Reply:
@atroyn is landing #129 which adds dimensionality metadata about the index. For now I think we can check on this. Once that lands can you update the logic here to utilize that?

Reply:
@HammadB sg will wait until #129 gets merged!

Reply:
Much appreciated!

Reply:
https://github.com/chroma-core/chroma/pull/129 is now merged.

Reply:
Sweet. Thanks @atroyn

Reply:
I think we're fine with this as is. It's going to be up to the user (for now) to pass in the right embedding function, but we have some guard rails around dimensionality at least. 

Happy to merge.